### DISCUSSION TITLE
useEffect double fetch

### DISCUSSION THREAD

**Discussion by Jeyhun1227** [vercel/next.js#59220] [Help]
### Summary

How can I prevent double fetch in useEffect in Next.js?

**Comment by aziyatali** [comment]
Hey, can provide more context on this?
There are several reasons why useEffect might fetching redundantly and we first need to figure out your case

**Comment by icyJoseph** [comment]
In React there's something called strict mode, which during development mode ONLY, triggers some work twice. In Effects, this happens even if dependencies remain unchanged.

- https://react.dev/reference/react/StrictMode
- https://legacy.reactjs.org/docs/strict-mode.html#ensuring-reusable-state from the legacy docs

You are most likely seeing this. React's strict mode here wants you to learn to, clean up when the effect is re-run. In your case, you have to learn/create a way to ignore a fetch that you no longer care about.

Personal shot out, but I wrote a little article showing how to abort fetch requests that are no longer relevant: https://medium.com/@icjoseph/using-react-to-understand-abort-controllers-eb10654485df

This is important because asynchronous work, such a network request, can take any amount of time, and end up in different states, so you have to account for these. In this case the unpredictable amount of time.

Say you have request 0, created by some dependencies on an effect, this request takes a long time, but then the user interacts with the page, and now you have different dependencies, so you make request 1, which happens to resolve almost immediately, you update the UI and all good. However, now request 0 resolves, and your code, sets state as it did with request 1, and it changes the UI. The user will be very confused and see the wrong information. You had to abort request 0 the moment the user made changes.
